internal/agent/reset.go

```diff
@@ -71,49 +139,59 @@ func Reset(reboot, unattended bool, dir ...string) error {
 
 	bus.Manager.Publish(sdk.EventBeforeReset, sdk.EventPayload{}) //nolint:errcheck
 
-	c, err := config.Scan(collector.Directories(dir...))
-	if err != nil {
-		return err
-	}
+	// Prepare a config from the cli flags
+	r := ExtraConfigReset{}
+	r.Reset.ResetOem = resetOem
 
-	utils.SetEnv(c.Env)
+	if resetOem {
+		r.Reset.ResetOem = true
+	}
 
-	// Load the installation Config from the cloud-config data
-	resetSpec, err := config.ReadResetSpecFromConfig(c)
-	if err != nil {
-		return err
+	if reboot {
+		r.Reset.Reboot = reboot
 	}
 
+	// Override the config with the event options
 	// Go over the possible options sent via event
 	if len(optionsFromEvent) > 0 {
 		if p := optionsFromEvent["reset-persistent"]; p != "" {
-			resetSpec.FormatPersistent = p == "true"
+			r.Reset.ResetPersistent = p == "true"
 		}
 		if o := optionsFromEvent["reset-oem"]; o != "" {
-			resetSpec.FormatOEM = o == "true"
-		}
-		if s := optionsFromEvent["strict"]; s != "" {
-			c.Strict = s == "true"
+			r.Reset.ResetOem = o == "true"
 		}
 	}
 
-	// Override with flags
-	if reboot {
-		resetSpec.Reboot = reboot
+	d, err := json.Marshal(r)
+	if err != nil {
+		c.Logger.Errorf("failed to marshal reset cmdline flags/event options: %s", err)
+		return c, err
 	}
+	cliConf := string(d)
 
-	err = resetSpec.Sanitize()
+	// cliconf goes last so it can override the rest of the config files
+	c, err = config.Scan(collector.Directories(dir...), collector.Readers(strings.NewReader(cliConf)))
 	if err != nil {
-		return err
+		return c, err
 	}
 
-	resetAction := action.NewResetAction(c, resetSpec)
-	if err := resetAction.Run(); err != nil {
-		fmt.Println(err)
-		os.Exit(1)
+	// Set strict validation from the event
+	if len(optionsFromEvent) > 0 {
+		if s := optionsFromEvent["strict"]; s != "" {
+			c.Strict = s == "true"
+		}
 	}
 
-	bus.Manager.Publish(sdk.EventAfterReset, sdk.EventPayload{}) //nolint:errcheck
+	utils.SetEnv(c.Env)
+
+	return c, nil
+}
 
-	return hook.Run(*c, resetSpec, hook.AfterReset...)
+// ExtraConfigReset is the struct that holds the reset options that come from flags and events
+type ExtraConfigReset struct {
```

Comment:
we should probably do the same on the upgrade.

I wonder if there may be any issues with overridable values...like we set some of this to false on the struct creation so maybe we are overriding values from the config with this ones....

Reply:
cc @mauromorales

Reply:
yeah I see your point, I would say we need to make sure we are setting them all every single time, and if we want to be extra careful then the logic to set them can go in one function that takes all 3